meta-iot2000-example/recipes-example/iot2000setup/files/expandfs.sh

```diff
@@ -1,31 +1,35 @@
 #!/bin/sh
 
-set -eu
+THIS_SCRIPT="$(basename $0)"
 
-PART_NUMBER=3
+log () {
+	echo ${THIS_SCRIPT}: $1 > /dev/kmsg
+}
 
-case $(mount | grep "on / ") in
-/dev/mmcblk0*)
-	ROOT_DEVICE="/dev/mmcblk0"
-	ROOT_PARTITION="/dev/mmcblk0p$PART_NUMBER"
-	;;
-/dev/sda*)
-	ROOT_DEVICE="/dev/sda"
-	ROOT_PARTITION="/dev/sda$PART_NUMBER"
-	;;
-*)
-	echo "Cannot determine boot device."
-	exit 1
-	;;
-esac
+disable () {
+	log "Deactivating this automatic resizer"
+	update-rc.d -f expandfs.sh remove
+}
 
-START_BLOCK=$(parted $ROOT_DEVICE -ms unit s p | grep "^$PART_NUMBER" | cut -f 2 -d: | sed "s/s$//")
+ROOT_DEVICE=$(findmnt / -o source -n | sed 's/p\?[0-9]*$//')
+LAST_PART=$(sfdisk -d ${ROOT_DEVICE} | tail -1 | awk '{print $1}')
 
-parted -ms $ROOT_DEVICE rm $PART_NUMBER
+log "Unmounting last partition"
+umount -f $LAST_PART > /dev/null
 
-parted -ms $ROOT_DEVICE unit s -- mkpart primary ext3 $START_BLOCK -1
+log "Resizing last partition"
+sfdisk -d ${ROOT_DEVICE} 2>/dev/null | grep -v last-lba | \
+	sed 's|\('${LAST_PART}' .*, \)size=[^,]*, |\1|' | \
+	sfdisk --force ${ROOT_DEVICE} 2>&1 | logger -t ${THIS_SCRIPT}
 
-partprobe $ROOT_DEVICE
-resize2fs $ROOT_PARTITION
+log "Informing kernel about new partitioning"
+partprobe > /dev/null
 
-update-rc.d -f expandfs.sh remove
+log "Resizing file system"
+resize2fs ${LAST_PART} 2>&1 | logger -t ${THIS_SCRIPT}
+
+log "Disabling this init script"
+disable
+
+log "Remounting last partition"
+mount $LAST_PART > /dev/null
```

Comment:
Does this script have dependencies on earlier patches? If not, then I would like to move it to the front, maybe pick it up earlier. May apply to other fixes / improvement patches as well.

Reply:
I am not quite sure if it works with the root partition currently mounted. I have not tested it, so I would prefer to take this together with the storage partition or any other last partition it should act on.

Reply:
The script already works in that mode, so does its brother in Isar.